gateway/src/main/kotlin/handler/HeartbeatHandler.kt

```diff
@@ -54,4 +54,14 @@ internal class HeartbeatHandler(
             ticker.stop()
         }
     }
+
+    // These two methods avoid a bug with the kotlinx.atomicfu compiler plugin
```

Comment:
'a bug' is not descriptive enough, if we have any issue on the matter reference to it, or reference to this PR.

Consider putting these comments on each function individually as well. Code has a history of drifting apart and being reorganized.

Reply:
I don't really know what the cause of this is though

Reply:
I was able to find the cause of one of the two issues: Kotlin/kotlinx.atomicfu#180 (will now link it from the code)

Reply:
Searching for the cause of the other bug I found actually two other bugs but could not track down which code introduces the bug i was looking for